File: PhysObjectExtractor/analysis/plot.py
import ROOT
f1 = ROOT.TFile('TT_10k.root')
t1 = f1.Get('myelectrons/Events')

fout = ROOT.TFile('hist.root','recreate')
h1 = ROOT.TH1F('h1',';number of high-pt electrons',6,-0.5,5.5)

for event in t1:
    nEle=0
    for ele_pt in event.electron_pt:
        if ele_pt > 5: nEle += 1
    h1.Fill(nEle)

fout.Write()
exit(0)

# c = ROOT.TCanvas()
# c.SetLogy()
# h1.SetLineColor(ROOT.kBlack)
# h1.Draw()
# c.SaveAs('example.pdf')

File: PhysObjectExtractor/analysis/plot.cxx
#include "TFile.h"
#include "TChain.h"
#include "TTreeReader.h"
#include "TTreeReaderArray.h"
#include "TH1F.h"

int main(){
    TChain *c1 = new TChain("myelectrons/Events");
    c1->Add("/eos/user/t/therwig/jtfi/240621_SingleNeutrino_prod1/output_*.root");
    //c1->Add("/eos/user/t/therwig/jtfi/240621_TT_prod1/output_*.root");
    TTreeReader reader(c1);
    TTreeReaderArray<float> ele_pt(reader, "electron_pt");
    
    TFile* fout = new TFile("hist.root","recreate");
    TH1F* h1 = new TH1F("h1",";number of high-pt electrons",6,-0.5,5.5);
    
    while (reader.Next()) {
        unsigned int nEle=0;
        for(unsigned int i=0; i < ele_pt.GetSize(); i++){
            if (ele_pt[i] > 5) nEle++;
        }
        h1->Fill(nEle);
    }
    fout->Write();
    return 0;
}
